jumpscale/core/base/store/etcd.py

```diff
@@ -0,0 +1,108 @@
+import etcd3
+import json
+
+from . import ConfigNotFound, EncryptedConfigStore, EncryptionMode
+
+from .serializers import JsonSerializer
+
+
+class EtcdStore(EncryptedConfigStore):
+    """
+    EtcdStore store is an EncryptedConfigStore
+
+    It saves the data in etcd and configuration for etcd comes from `config_env.get_store_config("etcd")`
+    """
+
+    def __init__(self, location):
+        """
+        create a new etcd store, the location given will be used to generate keys
+
+        this keys will be combined to get/set instance config
+
+        Args:
+            location (Location)
+        """
+        super().__init__(location, JsonSerializer())
+        etcd_config = self.config_env.get_store_config("etcd")
+        hostname = etcd_config.get("hostname", "localhost")
+        port = etcd_config.get("port", 2379)
+        self.etcd_client = etcd3.client(host=hostname, port=port)
+
+    def get_key(self, instance_name):
+        """
+        get a key for an instance
+
+        this will return a dot-separated key derived from current location
+
+        Args:
+            instance_name (str): name
+
+        Returns:
+            str: key
+        """
+        return ".".join([self.location.name, instance_name])
+
+    def read(self, instance_name):
+        """
+        read instance config from etcd
+
+        Args:
+            instance_name (name): name
+
+        Returns:
+            str: return json object
+        """
+        key = self.get_key(instance_name)
+        if not self.etcd_client.get(key)[0]:
+            raise ConfigNotFound(f"cannot find config for {instance_name} at {key}")
+
+        return json.loads(self.etcd_client.get(key)[0].decode())
```

Comment:
serialization would be taken care of using the given serializer, you can return the data as is.

Reply:
when we returned it as is it failed, because it is a string. @abom  does that make sense?

Reply:
I don't know why it failed, can you please give more details?

Reply:
I retired now with all your comments(removing the get as well) and it worked. Apparently the fail was because of the overriding of the get. Thanks, will update now